### DISCUSSION TITLE
Is there a way to include attachments stored in iCloud in the export?

### DISCUSSION THREAD

**Discussion by andrewmon13** [ReagentX/imessage-exporter#309] [Q&A] [answered]
I'm exporting about a years worth of iMessage data including attachments. It seems expanded attachments in my conversations are being exported correctly but the one's that aren't look like this in iMessage: 
<img width="316" alt="Screenshot 2024-07-30 at 1 06 49 AM" src="https://github.com/user-attachments/assets/6e584ef1-0e34-400d-9c59-d4ca4d365946">
These will display once I click on them.

Is there a way to include these in the export or expand them all in iMessage before exporting?

**Comment by ReagentX** [comment, marked as answer]
Those attachments are not stored locally, they're stored in iCloud. There is no way to programmatically download them; you have to manually download your attachments to export them. Luckily, both macOS and iOS have [buttons](https://support.apple.com/guide/messages/download-photos-videos-icloud-messages-mac-icht02671c2f/mac) to do this:

![image](https://github.com/user-attachments/assets/778095e5-abf2-4467-ae58-435c7e7d854d)
